Modules/Module_Processing/Module_ASL/xASL_wrp_VisualQC_ASL.m

```diff
@@ -134,12 +134,13 @@ function xASL_wrp_VisualQC_ASL(x)
     if xASL_exist(x.P.Path_c1T1,'file') && xASL_exist(x.P.Path_c2T1,'file')
         % If segmented files exist, then the PVgm should already be prepared previously
 
-        if ~xASL_exist(x.P.Path_PWI4D, 'file')
+        if ~isfield(x.P, 'Path_PWI4D_used') || ~xASL_exist(x.P.Path_PWI4D_used, 'file')
             [ControlIM, LabelIM] = xASL_quant_GetControlLabelOrder(xASL_io_Nifti2Im(x.P.Path_ASL4D));
-            xASL_io_SaveNifti(x.P.Path_ASL4D, x.P.Path_PWI4D, ControlIM-LabelIM);
+			x.P.Path_PWI4D_used = x.P.Path_PWI4D;
```

Comment:
I would not do this. We have a single management location for choosing which PWI4D path to use, and to perhaps revert to an older one. Such intermediate fixes to avoid the pipeline crashing are nice, but you have just created a potential bug!

Think about the situation that for whatever reason the PWI4D_used definition went wrong. Then, the user will look at the wrong visual QC! I would remove this, and in the previous instance where we define this path, we issue a warning.

Note that this also means that the merging and definition of PWI4D should also happen when the status file for quantification already exists. So these should happen just above the quantification part.

Reply:
The first two paragraphs - I agree, I merely worked what was already in the code. See the comment below.

Regarding the third part - I don't understand. The ASL module does first Resampling, the merging, then quantification. PWI4D merging has to happen before quantification.

Reply:
My point is that if the status files already exist, the definition of PWI4D_merged should still be done.

E.g., situation 1:
ASL module crashes after the quantification, and only the xASL_wrp_VisualQC is redone when restarting the ASL module (then we need to have the definition x.P.PWI4D_used redefined, even though it will skip the quantification because that status file already exists) -> so we move the PWI4D_used definition to before the status check for quantification

Reply:
Good point. Done.